File: repos_estudiantes_TARQUI_tercera_entrega/RUBIN_DE_CELIS_VALDEZ_ANDRES/PERSISTENCIA/EJERCICIO-5/Medicamento.py
class Medicamento:
    def __init__(self, nombre="", codigo=0, tipo="", precio=0.0):
        self.nombre = nombre
        self.codMedicamento = codigo
        self.tipo = tipo
        self.precio = precio
    
    def leer(self):
        self.nombre = input("Nombre del medicamento: ")
        self.codMedicamento = int(input("Código: "))
        self.tipo = input("Tipo (resfrio/tos/dolor): ")
        self.precio = float(input("Precio: "))
    
    def mostrar(self):
        print(f"Medicamento: {self.nombre}")
        print(f"Código: {self.codMedicamento}")
        print(f"Tipo: {self.tipo}")
        print(f"Precio: {self.precio:.2f}")
    
    def getTipo(self):
        return self.tipo
    
    def getPrecio(self):
        return self.precio
    
    def to_dict(self):
        return {
            'nombre': self.nombre,
            'codigo': self.codMedicamento,
            'tipo': self.tipo,
            'precio': self.precio
        }
    
    @classmethod
    def from_dict(cls, data):
        return cls(data['nombre'], data['codigo'], data['tipo'], data['precio'])

File: repos_estudiantes_ALVAREZ_tercera_entrega/HUANCA_CARDOZO_ADRIAN_GUSTAVO/PERSISTENCIA_DE_OBJETOS/EJERCICIO_5/Medicamento.java
package PERSISTENCIA_DE_OBJETOS.EJERCICIO_5;
import java.io.Serializable;

public class Medicamento implements Serializable {
    private String nombre;
    private int codMedicamento;
    private String tipo;
    private double precio;

    public Medicamento() {}

    public void leer(String nombre, int codMedicamento, String tipo, double precio) {
        this.nombre = nombre;
        this.codMedicamento = codMedicamento;
        this.tipo = tipo;
        this.precio = precio;
    }

    public void mostrar() {
        System.out.println("Nombre: " + nombre + ", Codigo: " + codMedicamento + ", Tipo: " + tipo + ", Precio: " + precio);
    }

    public String getTipo() {
        return tipo;
    }

    public double getPrecio() {
        return precio;
    }

    public String getNombre() {
        return nombre;
    }
}
